src/MongoDB.Driver.Core/Core/WireProtocol/CommandUsingCommandMessageWireProtocol.cs

```diff
@@ -186,7 +186,14 @@ private Type0CommandMessageSection<BsonDocument> CreateType0Section(ConnectionDe
             }
             Action<BsonWriterSettings> writerSettingsConfigurator = s => s.GuidRepresentation = GuidRepresentation.Unspecified;
 
-            if (_session.IsInTransaction)
+            string commandName = null;
+            if (_command.GetType() == typeof(BsonDocument))
+            {
+                commandName = _command.GetElement(0).Name;
+            }
+
+            _session.AboutToSendCommand(commandName);
+            if (_session.IsInTransaction || commandName == "commitTransaction")
```

Comment:
The conditional on line 200 comparing StatementId to 0.  That seems strange now that statementId is no longer a part of transactions.  Where is StatementId even being tracked now?

Reply:
I've removed StatementId.

The code that was using it now uses the State and IsEmpty properties instead.